pyuvsim/simsetup.py

```diff
@@ -526,7 +526,9 @@ def initialize_uvdata_from_params(obs_params):
         param_dict['object_name'] = '{}_ra{:.4f}_dec{:.4f}'.format(param_dict['sources']['catalog'], src.ra.deg, src.dec.deg)
     uv_obj = UVData()
     for k in param_dict:
-        if hasattr(uv_obj, k):
+        # use the __iter__ function on UVData to get list of UVParameters on UVData
+        param_name = [getattr(uv_obj, param).name for param in uv_obj]
+        if k in param_name:
```

Comment:
I think this block should come after line 549. This way, if the parameter file has `history` or `polarization_array`, etc. those won't be overwritten.

Reply:
ok, I'll make that change.

Reply:
although it looks like the history gets overwritten in any case in `init_uvdata_out` in `uvsim.py`

Reply:
Actually, all of the values that are hardcoded in lines 547-550 are overwritten in `init_uvdata_out`, so moving this won't change anything.

Reply:
That's true... I guess most development has neglected the situation when someone's simulating off a uvfits file and would want to keep such metadata, but I doubt that will come up often.

I'm starting to think we should deprecate support for simulation off of uvfits files. Each simulation needs more settings than can be given by a uvfits file, and so we'd need to maintain wrappers that provide defaults for those other settings or let the user set them by keyword or something. We have the machinery to make parameter files out of uvfits files anyway.

Reply:
I'm in favor of heading that direction. Make an issue and we can discuss it with the group.